pkg/utils/node.go

```diff
@@ -62,6 +63,47 @@ func GetNodeIP(node *apiv1.Node) (net.IP, error) {
 	return nil, errors.New("host IP unknown")
 }
 
+// GetNodeIP returns the most valid external facing IP address for a node (IPv4 and IPv6).
+// Order of preference:
+// 1. NodeInternalIP
+// 2. NodeExternalIP (Only set on cloud providers usually)
+func GetNodeIPDualStack(node *apiv1.Node, enableIPv4, enableIPv6 bool) (net.IP, net.IP, error) {
+	var ipAddrv4, ipAddrv6 net.IP
+	addresses := node.Status.Addresses
+	addressMap := make(map[apiv1.NodeAddressType][]apiv1.NodeAddress)
```

Comment:
Can we preallocate the size of this map with the length of the `addresses` slice? Also, will there be the chance anything in that slice could be nil?

Reply:
* Preallocating is a great idea!
* Which slice do you mean? `[]apiv1.NodeAddress`? Yes, that slice might be nil if the node doesn't have a particular IP type  (i.e. external). But I think we are fine in that case - later in this function we are checking if keys exist in the map and we are iterating over a slice. And we explicitly return an error if we don't find any address at all.

Reply:
Done the preallocating part.